### PR TITLE
Api 2064 enhance error logging

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
In response to many 500 errors not being handled by sentry and not having a clear path of resolution in Cloudwatch, I have enhanced the saml-proxy's error handling with the following changes:

- Errors with no status code will automatically be assumed to be 500's. This value will get assigned to these errors via middleware and sentry will be able to handle them.
- A unique request_id will be assigned to every request and included in the winston and morgan logs, so these errors can be traced in cloudwatch.
- error handling for empty `/sso` calls.

Research was done to justify these changes. For more context see the document below.
[Findings](https://github.com/department-of-veterans-affairs/vets-saml-proxy/files/5120154/API2064_Findings_And_Proposals.docx)

Tests to ensure proper functionality. 
[API2064 TestCases](https://github.com/department-of-veterans-affairs/vets-saml-proxy/files/5120155/API2064_TestCases.docx)

### CODE REVIEW

File: saml-proxy/src/routes/index.js [hunk 1]
```diff
@@ -8,15 +8,17 @@ import flash from 'connect-flash';
...
-import { loggingMiddleware, sassLogger } from '../logger';
+import { loggingMiddleware as morganMiddleware, winstonMiddleware, sassLogger, logger } from '../logger';
 import createPassport from "./passport";
 import addRoutes from "./routes";
 import configureHandlebars from "./handlebars";
 import { getParticipant } from "./handlers";
 
 import promBundle from 'express-prom-bundle';
 import * as Sentry from '@sentry/node';
+// import { logger } from "handlebars";
```
Comment (beaugrantham-va) [inline saml-proxy/src/routes/index.js:21]: Dead code

File: saml-proxy/src/routes/index.js [hunk 2]
```diff
@@ -143,6 +146,13 @@ export default function configureExpress(app, argv, idpOptions, spOptions, vetsA
 
   addRoutes(app, idpOptions, spOptions);
 
+  // Catches unhandled errors
+  app.use(function onError(err, req, res, next) {
+    err.status = err.status || 500;
+    logger.error("An unhandled error occured. ", err);
+    next(err)
```
Comment (beaugrantham-va) [inline saml-proxy/src/routes/index.js:153]: `;`

File: saml-proxy/src/routes/index.js [hunk 3]
```diff
@@ -154,8 +164,10 @@ export default function configureExpress(app, argv, idpOptions, spOptions, vetsA
     app.use(Sentry.Handlers.errorHandler({
       shouldHandleError(error) {
         if (error.status >= 400) {
+          logger.info("Error handled by sentry.");
```
Comment (beaugrantham-va) [inline saml-proxy/src/routes/index.js:167]: Are these meant to stick around? Or were they debugging lines?

**Comment by crolarlibertyva** [comment]
Some logging example in test run.
`[xmldom warning]        unclosed xml attribute 
@#[line:1,col:39]
=�٭9�]�����E�X���i\��ݡM�IYť=ٍ�M�(��Y<ɸ������=�a��%���%�dɡi-�9%坵��)�IYa�i�=e�Ye��]��4�ɬ�A�-�͜�����M������ɕY��Ք����-��%�������`����ф��G3��S�6W'F�f�6FS�Ԕ�GF�44ftt�$t�tu��"��t57u4�#4ET$7uTԔv$�7t5�EetWt�eW�UD�$Tp�TT4t�$g6u�f6�W�DUt�%tTT'wt�S$gT�U���sV���'�T��4tTT6wtUC'C�DUP��$�tTT7wt�S�T��fF�Ƶ�����D&t�d$��S%&�4�#%�3%��'��#�$�uc%�ww��D&v�����s�s$5UtEv�U�Ӕ#'C�3V�##t��4��Fwt�4�DW��DSUv�4�֦wt�4�DW��DSEv�4 �פT���tTT&��5edׄW�$&t�d$t�6��$v��#4�VtW�f�T&t�d$4�Ed�&�$v6�gU�&Ǡ��#��ED�&t�d$��$S�&DtW�dD4&t�d$4�3�EC'�#5��ug��'wtvu�EeDD$浥�#F�W@�F�c3&GfF��F�g4�'wtvu�������f4��$fs&էes�&DtWU�#�DԔ�$���&v����s�s �Td�4�Ԕ�$6t�4T�&�g3cv��7g�DT�D�%cu��v���R��7df��D4f�f6v�%"��p�3�GT�v����gT�f�Eft%�T�g�4g�B�g�6dG$2���b�F���WSf"����fEF�7T�F5v��s�@�Ӆ���TF�%gգd�SSWv�6����$�gf��t�FuF�Wf'$WE73t�F��''GF���d&Ĥ�#�'@�F#4V�WW�TD�3D&D�c�G7�6Ä�u6R��Զ%V�6G%t�E��T�fu�����s&��t$�G#���W6��F73��W�v��ƕ�4�fG6�Ĥd�3'#Vc5gSs���vFuGwsT�eg���&FT��u6�D$�t57p�4�#4ET$7uTD�$���v����v'Ef�w��et�b��Gs2�WU�t4t6Wd�v�Gg�S$���e�v�e��$�4u#�ֵGcW#�V��T�$��DUSd�g7s�TV�fuE7u4�t׃gc�7r��tT�u6�G6���5#'�6�`�f�w77��$eF$$�����D�tSV��vԇ��S�F�דD�F�ӕ�4#t�W�'&���%3Ut����$�4s������d2�'44UF�fע�%��ƶ�E�&��5D�4S���7Wf��V�wf��&G�fT�s��Se�Ft��6ЦD�4�464g��&�Vׅ�eV�d�e�6��ccGdd��'r�G6�#f�W�%4����G3��S�6W'F�f�6FS`O
                                                                                                                                                                                �LQ]O�
     �Β�^R[��́</ds:Signature6�ͅ�����9���%A������ɵ����ɸ齅ͥ�鹅����ь�M50�ĸ�鹅�������ɵ���չ�������������ͅ������ѡ�I��Օ���&RelayState=%252Foauth2%252Fv1%252Fauthorize%252Fredirect%253Fokta_key%253Dy21U0ud-LwgVMDRnJvOfqwSlP8LQ-EtH9XL1BNQMNWM is not complete:saml2:Issuer
{
  "service": "saml-proxy",
  "request_id": "647312f0-e88c-11ea-ae16-4fb47269d680",
  "action": "parseSamlRequest",
  "result": "success",
  "session": "f43d8369-8205-4f46-bdd7-a40022b18a7c",
  "level": "info",
  "message": "User arrived from Okta. Rendering IDP login template.",
  "time": "2020-08-27T17:40:26.722Z"
}`

File: saml-proxy/src/routes/index.js (hunk 3 above)
Comment (lukemarkwordtlibertyits) [reply saml-proxy/src/routes/index.js:167]: I intentionally left these logs in here, but was curious about your thoughts. For debugging they are extremely helpful, but I thought they would also be helpful if we have any other errors that are behaving funny in Sentry (i.e. not getting logged like our previous `500`s). I could go either way. Thoughts?

Comment (beaugrantham-va) [reply saml-proxy/src/routes/index.js:166]: I'm fine with keeping them for now. If they prove to be more spam than useful we can drop them.

**Pull request merged**